src/dict_list.jl

```diff
@@ -275,3 +298,85 @@ julia> dict_list(d) # only in case `:a` is `1` the dictionary will get key `:c`
      end
      :(toDependentParameter($(esc(value)),$(esc(condition))))
  end
+
+"""
+   Derived(parameters :: Vector{Union{String,Symbol}},function :: Function)
+Wrap the name(s) of a parameter(s) and a function. After the 
+possible parameter combinations are created, dict_list will replace instances of 
+Derived by the result of the function func, evaluated with the value of 
+the parameter(s).
+
+## Examples
+```
+julia> p = Dict(:α => [1, 2],
+           :solver => ["SolverA","SolverB"],
+           :β => Derived(:α, x -> x^2),
+           )
+Dict{Symbol, Any} with 3 entries:
+  :α      => [1, 2]
+  :solver => ["SolverA", "SolverB"]
+  :β      => Derived{Symbol}(:α, #51)
+
+julia> dict_list(p)
+4-element Vector{Dict{Symbol, Any}}:
+ Dict(:α => 1, :solver => "SolverA", :β => 1)
+ Dict(:α => 2, :solver => "SolverA", :β => 4)
+ Dict(:α => 1, :solver => "SolverB", :β => 1)
+ Dict(:α => 2, :solver => "SolverB", :β => 4)
+```
+A vector of parameter names can also be passed when the accompanying function 
+uses multiple arguments:
+```julia
+ julia> p2 = Dict(:α => [1, 2],
+           :β => [10,100],
+           :solver => ["SolverA","SolverB"],
+           :γ => Derived([:α,:β], (x,y) -> x^2 + 2y),
+           )
+Dict{Symbol, Any} with 4 entries:
+  :α      => [1, 2]
+  :γ      => Derived{Symbol}([:α, :β], #7)
+  :solver => ["SolverA", "SolverB"]
+  :β      => [10, 100]
+
+julia> dict_list(p2)
+8-element Vector{Dict{Symbol, Any}}:
+ Dict(:α => 1, :γ => 21, :solver => "SolverA", :β => 10)
+ Dict(:α => 2, :γ => 24, :solver => "SolverA", :β => 10)
+ Dict(:α => 1, :γ => 21, :solver => "SolverB", :β => 10)
+ Dict(:α => 2, :γ => 24, :solver => "SolverB", :β => 10)
+ Dict(:α => 1, :γ => 201, :solver => "SolverA", :β => 100)
+ Dict(:α => 2, :γ => 204, :solver => "SolverA", :β => 100)
+ Dict(:α => 1, :γ => 201, :solver => "SolverB", :β => 100)
+ Dict(:α => 2, :γ => 204, :solver => "SolverB", :β => 100)
+```
+"""
+struct Derived{T}
+    independentParam::Vector{T}
+    func::Function
+end
+
+"""
+    Derived(independentP <: Union{String,Symbol}, func :: Function)
+Constructs a Derived from a single independent parameter.
+"""
+function Derived(independentP :: Union{String,Symbol}, func :: Function) 
```

Comment:
weird syntax to have white space between `::`. I/ve never seen it before. Should be without white space.